File: backend/models/hospital.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const hospitalSchema = new Schema({
  kode_rumah_sakit: String,
  nama_rumah_sakit: String,
  jenis_rumah_sakit: String,
  alamat_rumah_sakit: String,
  kelurahan: String,
  kecamatan: String,
  kota_administrasi: String,
  kode_pos: String,
  nomor_telepon: String,
  website: String,
  email: String,
});

const Hospital = mongoose.model("Hospital", hospitalSchema);
exports.model = Hospital;
exports.schema = hospitalSchema;

File: frontend/lib/models/hospital.dart
class HospitalModel {
  final String id;
  final String name;
  final String type;
  final String address;
  final String city;
  final String postalCode;
  final String phoneNumber;
  final String website;
  final String email;

  HospitalModel({
    required this.id,
    required this.name,
    required this.type,
    required this.address,
    required this.city,
    required this.postalCode,
    required this.phoneNumber,
    required this.website,
    required this.email,
  });

  factory HospitalModel.fromJSON(Map<String, dynamic> json) {
    return HospitalModel(
      id: json['kode_rumah_sakit'],
      name: json['nama_rumah_sakit'],
      type: json['jenis_rumah_sakit'],
      address:
          '${json['alamat_rumah_sakit'] == '' ? '' : json['alamat_rumah_sakit']}, ${json['kelurahan'] == '' ? '' : json['kelurahan']}, ${json['kecamatan'] == '' ? '' : json['kecamatan']}',
      city: json['kota_administrasi'] == '' ? '' : json['kota_administrasi'],
      postalCode: json['kode_pos'] == '' ? '' : json['kode_pos'],
      phoneNumber: json['nomor_telepon'] == '' ? '' : json['nomor_telepon'],
      website: json['website'] == '' ? '' : json['website'],
      email: json['email'] == '' ? '' : json['email'],
    );
  }
}
